### DISCUSSION TITLE
Please register "uws" or "uwebsockets" package name on npmjs...

### DISCUSSION THREAD

**Discussion by ozra** [uNetworking/uWebSockets.js#92] [Q&A] [answered]
...before someone else does it. 

Lowercase (hyphenated) packages is de facto norm, and being able to simply use `yarn add uws`, `yarn add uwebsockets` / `npm i -S uwebsockets` from the registry will promote the project I'd think.

Or "@unetworking/*", if you intend to add a lot of different js modules that compose well together — but I think "uws" would rock.

**Comment by ghost** [comment]
I did that three years ago.

**Comment by ozra** [comment]
xD

**Comment by iMoses** [comment]
@alexhultman and it was fabulous! but why is it not used today for the new versions of the library?

https://www.npmjs.com/package/uws

I'm not sure why this library can only be installed from github which doesn't inform me about any new versions, which doesn't respect the semver logic of `package.json`, which is basically very patchy.. Is there an official reason why it isn't in the npm registry?

**Comment by ghost** [comment, marked as answer]
I dislike NPM very, very much. For many reasons.

**Reply by iMoses** [reply]
> I dislike NPM very, very much. For many reasons.

LOL I believe you :) 

But is there a specific reason? Maybe something any of us can help with? If it's an issue with NPM then maybe a large enough number of complaints can move things along...

**Reply by ianzone** [reply]
I still prefer to use the npm registry, cuz github connection is very slow from my place, and the current method installs all binaries for all platforms, which is space and time-consuming. If you agree, I can make a pr to auto release npm packages.

**Reply by paulius-valiunas** [reply]
I would like to hear your reasons.

**Comment by uNetworkingAB** [comment]
There are PRs for only installing the binary of the host system but I
haven't had the time to look into it. Also, you are 9 years too late on
this question.

Den fre 7 feb. 2025 18:57Yiheng ***@***.***> skrev:
